### PR TITLE
feat: Dexscreener fetcher

### PR DESCRIPTION
**Pull request by akhercha**
Closes #NA

## Introduced changes
- Added a Dexscreener fetcher 
- Renamed StarknetAMM to Ekubo

##

- [x] This PR contains breaking changes

StarknetAMM source name is now EKUBO.

### CODE REVIEW

File: pragma-sdk/pragma_sdk/common/fetchers/fetchers/dexscreener.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            result=float(pair_data["priceUsd"]),
+            volume=float(pair_data["volume"]["h24"]),
+        )
+
+    async def _query_dexscreener(
+        self, base: Currency, quote: Currency, session: ClientSession
+    ) -> dict | PublisherFetchError:
+        pair_id = f"{base.id}/{quote.id}"
+        url = self.format_url(Pair.from_tickers(base.id, quote.id))
+        print(url)
```
Comment (EvolveArt) [inline pragma-sdk/pragma_sdk/common/fetchers/fetchers/dexscreener.py:114]: typo

File: pragma-sdk/pragma_sdk/common/fetchers/fetchers/dexscreener.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        self,
+        pair: Pair,
+        session: ClientSession,
+    ) -> SpotEntry | PublisherFetchError:
+        """
+        Query the dexscreener API and construct the SpotEntry.
+
+        NOTE: It is really unclear at the moment how the pair is actually constructed,
+        sometimes the quote asset is in front of the base asset...
+        To be sure it works, we try both.
```
Comment (EvolveArt) [inline pragma-sdk/pragma_sdk/common/fetchers/fetchers/dexscreener.py:85]: but if we try both then the price won't be the same, like ETH/USDC != USDC/ETH

File: pragma-sdk/pragma_sdk/common/types/currency.py [hunk 3]
```diff
@@ -17,8 +17,8 @@ def __init__(
         currency_id: str,
         decimals: Decimals,
         is_abstract_currency: bool,
-        starknet_address: Optional[Address] = None,
-        ethereum_address: Optional[Address] = None,
+        starknet_address: Optional[int | str] = None,
+        ethereum_address: Optional[int | str] = None,
```
Comment (EvolveArt) [inline pragma-sdk/pragma_sdk/common/types/currency.py:21]: just change the `Address` type definition then

**Review by EvolveArt — CHANGES_REQUESTED** [review]

File: pragma-sdk/pragma_sdk/common/fetchers/fetchers/dexscreener.py (hunk 2 above)
Comment (akhercha) [reply pragma-sdk/pragma_sdk/common/fetchers/fetchers/dexscreener.py:85]: I agree, but for example for EKUBO the pair EKUBO/USDC as addresses is formatted the other way around and it returns the EKUBO/USDC price

File: pragma-sdk/pragma_sdk/common/types/currency.py (hunk 3 above)
Comment (akhercha) [reply pragma-sdk/pragma_sdk/common/types/currency.py:21]: It's fine to have only int, but we can construct it from str or int

File: pragma-sdk/pragma_sdk/common/fetchers/fetchers/dexscreener.py (hunk 2 above)
Comment (EvolveArt) [reply pragma-sdk/pragma_sdk/common/fetchers/fetchers/dexscreener.py:85]: just checked, weird but true indeed

**Review by EvolveArt — APPROVED** [review]

**Pull request merged**